src/Modules/Render/ViewScene.cc

```diff
@@ -51,20 +51,31 @@ using namespace SCIRun::Core::Logging;
 MODULE_INFO_DEF(ViewScene, Render, SCIRun)
 
 Mutex ViewScene::mutex_("ViewScene");
+//Mutex ViewScene::asyncExecuteMutex_("ViewSceneAsyncExecuteMutex");
 
 ALGORITHM_PARAMETER_DEF(Render, GeomData);
+ALGORITHM_PARAMETER_DEF(Render, VSMutex);
 ALGORITHM_PARAMETER_DEF(Render, GeometryFeedbackInfo);
 ALGORITHM_PARAMETER_DEF(Render, ScreenshotData);
 ALGORITHM_PARAMETER_DEF(Render, MeshComponentSelection);
 ALGORITHM_PARAMETER_DEF(Render, ShowFieldStates);
 
-ViewScene::ViewScene() : ModuleWithAsyncDynamicPorts(staticInfo_, true), asyncUpdates_(0)
+ViewScene::ViewScene() : ModuleWithAsyncDynamicPorts(staticInfo_, true)
 {
   RENDERER_LOG_FUNCTION_SCOPE;
   INITIALIZE_PORT(GeneralGeom);
   INITIALIZE_PORT(ScreenshotDataRed);
   INITIALIZE_PORT(ScreenshotDataGreen);
   INITIALIZE_PORT(ScreenshotDataBlue);
+
+  get_state()->setTransientValue(Parameters::VSMutex, new Mutex("ViewSceneScreenShotMutex"), true);
```

Comment:
No raw pointers/`new` allowed. This can be a `std::unique_ptr`, or just store it as an instance variable value and pass an address to the state.